python/ray/data/_internal/logging.py

```diff
@@ -206,3 +214,45 @@ def get_log_directory() -> Optional[str]:
 
     session_dir = global_node.get_session_dir_path()
     return os.path.join(session_dir, "logs", "ray-data")
+
+
+def get_default_formatter() -> logging.Formatter:
+    log_encoding = os.environ.get(RAY_DATA_LOG_ENCODING_ENV_VAR_NAME)
+    if log_encoding is not None and log_encoding.upper() == "JSON":
+        return DEFAULT_JSON_FORMATTER()
+
+    return logging.Formatter(DEFAULT_TEXT_FORMATTER)
+
+
+def register_dataset_logger(dataset_id: str) -> None:
+    """Add a logger for a dataset with the given ID.
+
+    Args:
+        dataset_id: The ID of the dataset.
+    """
+    logger = logging.getLogger("ray.data")
+    handler = SessionFileHandler(
+        filename=f"ray-data-{dataset_id}.log",
+    )
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(get_default_formatter())
+
+    # Add the handler to the logger
+    logger.addHandler(handler)
+    DATASET_LEVEL_LOGGERS[dataset_id] = handler
```

Comment:
I like this idea. But there is an issue.
It's not common, but technically you can run multiple datasets concurrently with multi-threading. 
In that case, the logger will be incorrectly overridden. 
One solution I thought of is to use a thread local handler. But the problem is some ray data functions leverage multi-threading, we'll also need to make sure the thread local variable propagate correctly to the sub threads.